GRT 0.0.1 (new formula)

nebgnahz opened the pull request:
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

-----

The Gesture Recognition Toolkit (GRT) for Real-time machine learning,
developed by Nick Gillian.

mikemcquaid commented on Formula/grt.rb:
```diff
@@ -0,0 +1,29 @@
+    system "make", "-j#{Hardware::CPU.cores}"
+    system "make", "install"
+  end
+  test do
```
Newline above this line.

mikemcquaid commented on Formula/grt.rb:
```diff
@@ -0,0 +1,29 @@
+    mkdir "build"
+    cd "build"
+    system "cmake", "-DCMAKE_INSTALL_PREFIX:PATH=#{prefix}", ".."
+    system "make", "-j#{Hardware::CPU.cores}"
```
You don't need the `-j`; this is done by default.

mikemcquaid commented on Formula/grt.rb:
```diff
@@ -0,0 +1,29 @@
+  def install
+    cd "build"
+    mkdir "build"
+    cd "build"
```
Use `mkdir "build" do` block instead

mikemcquaid commented on Formula/grt.rb:
```diff
@@ -0,0 +1,29 @@
+    cd "build"
+    mkdir "build"
+    cd "build"
+    system "cmake", "-DCMAKE_INSTALL_PREFIX:PATH=#{prefix}", ".."
```
Use `std_cmake_args` here instead of specifying `prefix`

mikemcquaid commented on Formula/grt.rb:
```diff
@@ -0,0 +1,29 @@
+    system "make", "install"
+  end
+  test do
+    # A simple test to see if we have GRT setup properly (only compilation).
```
Can remove this comment.

mikemcquaid commented on Formula/grt.rb:
```diff
@@ -0,0 +1,29 @@
+class Grt < Formula
+  desc "The Gesture Recognition Toolkit (GRT) for Real-time machine learning."
+  homepage "http://www.nickgillian.com/wiki/"
```
http://www.nickgillian.com/wiki/pmwiki.php/GRT/GestureRecognitionToolkit ?

nebgnahz commented:
Fixed based on the comments (I guess no need to reply each individually). I am reproducing my commit message below:

Fixes based on the review.
1. New line before `test do` block.
2. No `-j` in `make`.
3. Use `mkdir [] do` block.
4. Use `std_cmake_args`.
5. Remove comments.
6. Updated the URL.

UniqMartin commented on Formula/grt.rb:
```diff
@@ -0,0 +1,29 @@
+        return 0;
+      }
+    EOS
+    system ENV.cc, "test.cpp", "-L#{lib}", "-lgrt", "-o", "test"
```
`test.cpp` looks like a C++ program (and has a matching extension), but you're using the C compiler (`ENV.cc`) to compile the file. Please use `ENV.cxx` instead (to reduce confusion and avoid problems in the future).

UniqMartin commented on Formula/grt.rb:
```diff
@@ -0,0 +1,29 @@
+class Grt < Formula
+  desc "The Gesture Recognition Toolkit (GRT) for Real-time machine learning."
```
These are minor stylistic issues, but please:

- Drop `(GRT)`. The formula name `grt` will be printed in virtually any context where this description will appear.
- Why is “Real-time” capitalized? I don't think it should be.
- Drop the full stop at the end.

UniqMartin commented on Formula/grt.rb:
```diff
@@ -0,0 +1,29 @@
+    cd "build"
+    mkdir "build" do
+      system "cmake", "..", *std_cmake_args
+      system "make"
```
Please remove this line. `make` is implied by `make install` and builds before installing unless there are issues with the build system (unlikely if it is generated by CMake).

nebgnahz commented:
@UniqMartin Thanks for the feedback. All are fixed.

PTAL.

UniqMartin commented:
Merged in d43a097c9f503410abb784fcf75bd3d784120977. Thank you for this contribution to Homebrew, @nebgnahz! :tada:
